Use preact-modal in adminModal

rt4914 opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://developers.forem.com/contributing-guide/forem#create-a-pull-request
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR replaces regular modal with preact-modal in adminModal file.

## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #16328
- Closes #16328

## QA Instructions, Screenshots, Recordings

**How to test this?**
1. Sign in as admin and visit http://localhost:3000/admin/customization/config
2. Click on `Authentication`
3. Click `Enable`, do not enter keys.
4. Click `Update Settings`
5. A modal will appear with warning


All three-instances where modal is getting used.

<img width="600" alt="Screenshot 2022-10-25 at 2 17 58 PM" src="https://user-images.githubusercontent.com/9396084/197728471-618aba8a-d608-464c-bf8d-3236c7c74650.png">

<img width="600" alt="Screenshot 2022-10-26 at 2 26 25 PM" src="https://user-images.githubusercontent.com/9396084/197982127-8147a9a2-c4d6-4f44-b717-2c085076c447.png">

<img width="600" alt="Screenshot 2022-10-26 at 2 26 41 PM" src="https://user-images.githubusercontent.com/9396084/197982143-0cce6a76-b901-42b8-8e2a-9e657a449024.png">


### UI accessibility concerns?

None

## Added/updated tests?

- [x] Yes
- [ ] No
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

rt4914 commented on app/javascript/admin/__tests__/controllers/config_controller.test.js:
```diff
@@ -1,34 +0,0 @@
-import { Application } from '@hotwired/stimulus';
-import ConfigController from '../../controllers/config_controller';
-
-describe('ConfigController', () => {
```
Removed this test file as we are using preact-modal now and similar to other instances I have added cypress tests for checking this entire functionality.

fdocr reviewed (changes requested):
Thanks for this @rt4914, this modal looks great and it's awesome to see we're closing these loops (refactors) 🙌🏼 

I did find it impossible to confirm the modal actions though (enable/disable auth providers). Here's a screen recording of what I see. I did notice the Cypress tests didn't include a flow that ensures confirming the modal actually take the action requested. Perhaps I'm missing something?

https://user-images.githubusercontent.com/6045239/197902746-e32abee8-1265-48fb-b7be-939584293224.mov

rt4914 commented:
> Thanks for this @rt4914, this modal looks great and it's awesome to see we're closing these loops (refactors) 🙌🏼
> 
> I did find it impossible to confirm the modal actions though (enable/disable auth providers). Here's a screen recording of what I see. I did notice the Cypress tests didn't include a flow that ensures confirming the modal actually take the action requested. Perhaps I'm missing something?
> 
>  confirmation_modal_screen_recording.mov

@fdocr Yes you are right that this feature was missing. I have made following changes:
1. fixed the bug
2. added screenshot of dialogs getting used al all-three places in code
3. added tests to make sure the disable-dialog is also covered.

Thanks.

fdocr reviewed (approved):
Tested and works great. Thanks @rt4914!
